Marlin/src/module/endstops.cpp

```diff
@@ -288,10 +288,24 @@ void Endstops::not_homing() {
   void Endstops::enable_z_probe(const bool onoff) {
     z_probe_enabled = onoff;
 
-    update();
+    resync();
   }
 #endif
 
+// Be sure to synchronize endstops status when enabled
+void Endstops::resync() {
+  if (!abort_enabled()) return;     // If endstops/probes are disabled the loop below can hang
+
+  #if ENABLED(ENDSTOP_INTERRUPTS_FEATURE) && !ENDSTOP_NOISE_THRESHOLD
+    update();
+  #else
+    safe_delay(2);  // Wait for Temperature ISR (runs at 1KHz)
```

Comment:
Since all the temperature ISR does is call 'poll' why not just call 'poll'?

Reply:
Interrupt/no threshold:
- `update` will update status and is not called by temperature ISR
Interrupt + threshold:
- `update` is not called by temperaturte ISR because of `endstop_poll_count` is 0 (we come from a disabled state so for sure it's 0). We need to start detection and then let temperature ISR filter value
poll with/without threshold:
- interrupt ISR will call update that will acquire correct value (I may also call directly update here)
  but I need at least first call to be sure it initialize `endstop_poll_count` (when active)